common/persistence/persistence-factory/factory.go

```diff
@@ -305,29 +296,37 @@ func (f *factoryImpl) getCassandraConfig() *config.Cassandra {
 	return cfg.DataStores[cfg.VisibilityStore].Cassandra
 }
 
-func newStore(cfg config.DataStore, tb tokenbucket.TokenBucket, clusterName string, maxConnsOverride int, logger log.Logger) Datastore {
-	var ds Datastore
-	ds.ratelimit = tb
-	if cfg.SQL != nil {
-		ds.factory = newSQLStore(*cfg.SQL, clusterName, maxConnsOverride, logger)
-		return ds
-	}
-	ds.factory = newCassandraStore(*cfg.Cassandra, clusterName, maxConnsOverride, logger)
-	return ds
-}
+func (f *factoryImpl) init(clusterName string, limiters map[string]tokenbucket.TokenBucket) {
+	f.datastores = make(map[storeType]Datastore, len(storeTypes))
+	defaultCfg := f.config.DataStores[f.config.DefaultStore]
+	defaultDataStore := Datastore{ratelimit: limiters[f.config.DefaultStore]}
+	switch {
```

Comment:
can we add a check if both defaultCfg.Cassandra and defaultCfg.SQL are not nil?

Reply:
that's done at config.Validate() at initialization time. Over here, its a fatal condition and the default case fatals